src/pineko/comparator.py

```diff
@@ -32,19 +35,66 @@ def compare(pine, fktable, max_as, max_al, pdf, xir, xif):
     """
     import lhapdf  # pylint: disable=import-error,import-outside-toplevel
 
-    pdfset = lhapdf.mkPDF(pdf, 0)
-    pdgid = int(pdfset.set().get_entry("Particle"))
+    pdfset1 = lhapdf.mkPDF(pdf1, 0)
+    if pdf2 is not None:
+        pdfset2 = lhapdf.mkPDF(pdf2, 0)
+    else:
+        pdfset2 = pdfset1
+
+    # TODO: This should probably changed in the future to use the Grid::convolutions
+    try:
+        parton1 = int(pine.key_values()["convolution_particle_1"])
+        parton2 = int(pine.key_values()["convolution_particle_2"])
+
+        if (
+            fktable.key_values()["convolution_type_1"]
+            != pine.key_values()["convolution_type_1"]
+            or fktable.key_values()["convolution_type_2"]
+            != pine.key_values()["convolution_type_2"]
+        ):
+            raise TypeError("Grids and FkTables do not have the same convolution types")
```

Comment:
I believe what @scarlehoff meant was
```suggestion
            raise ValueError(f"Grids and FkTables do not have the same convolution types: grid=({pine.key_values()["convolution_type_1"]},{pine.key_values()["convolution_type_2"]}) vs. fk=({fktable.key_values()["convolution_type_1"]},{fktable.key_values()["convolution_type_2"]})")
```
(typing in the void so watch out) (also we maybe want some new lines)

Reply:
[8642572](https://github.com/NNPDF/pineko/pull/181/commits/8642572c3188a4f7264bc7a89f8c1df5fe155ff1)